cschlaffner/PROTzilla2
# Single protein

## Description

- The goal of this pr is to enable mapping between the the protein df and the peptide/evidence df by aligning the 'Protein ID' column
- The 'Protein ID' column now contains a single protein id for each entry in both the protein df as well as peptide/evidence df
- For the protein df the first valid Protein of the 'Majority Protein' column is selected 
- For the peptide/evidence df the leading razor Protein is selected
- These changes make it possible to map at least one peptide of the peptide df for every protein in the protein df 
- Should a future step need the complete Protein Groups they can easily be added as an additional column 'Protein Group'

## Changes

- The importing Methods where changed to select the new columns
- clean Protein Groups was changed to retain the initial order of the Protein Group and then select the first of the Group 
- Tests where changed to accommodate the new behavior

## Testing

- Import a protein df and compare the read with the resulting dataframe
- Execute the 'Filter Peptides of Protein' method. It should select at least one peptide for every protein you select

## PR checklist
**Development**
 
**Mergeability**
- [ ] The code has been formatted with `black`
 
**Code review**
